include/iterlib/LetIterator.h

```diff
@@ -0,0 +1,54 @@
+// Copyright 2015 - present, Facebook
+
+#pragma once
+
+#include "iterlib/Iterator.h"
+#include "iterlib/WrappedIterator.h"
+
+namespace iterlib {
+
+class LetIterator : public WrappedIterator {
+ public:
+  explicit LetIterator(Iterator *inner,
+                       dynamic newKey,
+                       dynamic oldKey)
+      : WrappedIterator(inner)
+      , newKey_(std::move(newKey))
+      , oldKey_(std::move(oldKey)) {}
+
+  const Item& value() const override {
+    auto omap = variant::ordered_map_t{};
```

Comment:
Shouldn't this be abstracted as a deep clone function in iterlib::dynamic itself?